src/validate.js

```diff
@@ -162,22 +205,36 @@ export default function validateFormData(
 
   let errors = transformAjvErrors(ajv.errors);
 
-  if (typeof transformErrors === "function") {
+  if (
+    typeof transformErrors === "function" &&
+    typeof customValidate !== "function"
+  ) {
     errors = transformErrors(errors);
   }
-  const errorSchema = toErrorSchema(errors);
 
   if (typeof customValidate !== "function") {
+    const errorSchema = toErrorSchema(errors);
     return { errors, errorSchema };
   }
 
+  // Apply custom validations and allow transform of merged result
   const errorHandler = customValidate(formData, createErrorHandler(formData));
   const userErrorSchema = unwrapErrorHandler(errorHandler);
-  const newErrorSchema = mergeObjects(errorSchema, userErrorSchema, true);
+  const userErrors = flattenErrorSchema(userErrorSchema).map(error =>
```

Comment:
Is there a reason why you're flattening & merging regardless of whether `transformErrors` is defined (as opposed to doing so only when `transformErrors` is defined)?

Reply:
I don't think so, I'll conditionally do this